types/interfaces.d.ts

```diff
@@ -0,0 +1,1796 @@
+    sensor_msgs.srv.SetCameraInfo_Response |
+    shape_msgs.msg.Mesh |
+    shape_msgs.msg.MeshTriangle |
+    shape_msgs.msg.Plane |
+    shape_msgs.msg.SolidPrimitive |
+    std_msgs.msg.Bool |
+    std_msgs.msg.Byte |
+    std_msgs.msg.ByteMultiArray |
+    std_msgs.msg.Char |
+    std_msgs.msg.ColorRGBA |
+    std_msgs.msg.Empty |
+    std_msgs.msg.Float32 |
+    std_msgs.msg.Float32MultiArray |
+    std_msgs.msg.Float64 |
+    std_msgs.msg.Float64MultiArray |
+    std_msgs.msg.Header |
+    std_msgs.msg.Int16 |
+    std_msgs.msg.Int16MultiArray |
+    std_msgs.msg.Int32 |
+    std_msgs.msg.Int32MultiArray |
+    std_msgs.msg.Int64 |
+    std_msgs.msg.Int64MultiArray |
+    std_msgs.msg.Int8 |
+    std_msgs.msg.Int8MultiArray |
+    std_msgs.msg.MultiArrayDimension |
+    std_msgs.msg.MultiArrayLayout |
+    std_msgs.msg.String |
+    std_msgs.msg.UInt16 |
+    std_msgs.msg.UInt16MultiArray |
+    std_msgs.msg.UInt32 |
+    std_msgs.msg.UInt32MultiArray |
+    std_msgs.msg.UInt64 |
+    std_msgs.msg.UInt64MultiArray |
+    std_msgs.msg.UInt8 |
+    std_msgs.msg.UInt8MultiArray |
+    std_srvs.srv.Empty_Request |
+    std_srvs.srv.Empty_Response |
+    std_srvs.srv.SetBool_Request |
+    std_srvs.srv.SetBool_Response |
+    std_srvs.srv.Trigger_Request |
+    std_srvs.srv.Trigger_Response |
+    stereo_msgs.msg.DisparityImage |
+    tf2_msgs.action.LookupTransformAction |
+    tf2_msgs.action.LookupTransformActionFeedback |
+    tf2_msgs.action.LookupTransformActionGoal |
+    tf2_msgs.action.LookupTransformActionResult |
+    tf2_msgs.action.LookupTransformFeedback |
+    tf2_msgs.action.LookupTransformGoal |
+    tf2_msgs.action.LookupTransformResult |
+    tf2_msgs.msg.LookupTransformAction |
+    tf2_msgs.msg.LookupTransformActionFeedback |
+    tf2_msgs.msg.LookupTransformActionGoal |
+    tf2_msgs.msg.LookupTransformActionResult |
+    tf2_msgs.msg.LookupTransformFeedback |
+    tf2_msgs.msg.LookupTransformGoal |
+    tf2_msgs.msg.LookupTransformResult |
+    tf2_msgs.msg.TF2Error |
+    tf2_msgs.msg.TFMessage |
+    tf2_msgs.srv.FrameGraph_Request |
+    tf2_msgs.srv.FrameGraph_Response |
+    trajectory_msgs.msg.JointTrajectory |
+    trajectory_msgs.msg.JointTrajectoryPoint |
+    trajectory_msgs.msg.MultiDOFJointTrajectory |
+    trajectory_msgs.msg.MultiDOFJointTrajectoryPoint |
+    unique_identifier_msgs.msg.UUID |
+    visualization_msgs.msg.ImageMarker |
+    visualization_msgs.msg.InteractiveMarker |
+    visualization_msgs.msg.InteractiveMarkerControl |
+    visualization_msgs.msg.InteractiveMarkerFeedback |
+    visualization_msgs.msg.InteractiveMarkerInit |
+    visualization_msgs.msg.InteractiveMarkerPose |
+    visualization_msgs.msg.InteractiveMarkerUpdate |
+    visualization_msgs.msg.Marker |
+    visualization_msgs.msg.MarkerArray |
+    visualization_msgs.msg.MenuEntry |
+    visualization_msgs.srv.GetInteractiveMarkers_Request |
+    visualization_msgs.srv.GetInteractiveMarkers_Response;
+}
```

Comment:
I have a question here, what's going to happen if a user creates a `.msg`/`.srv` defined by himself? how can offer the type information?

Reply:
My understanding is that when a user creates a custom msg or srv, he must run rclnodejs.regenerateAll() to generate the new js structures. But unfortunately I did not account for regenerateAll() to additionally generate a new interfaces.d.ts updated with new type info for the new msg or srv. My BAD as I only modified the scripts/generate_messages.js#generateAll() to generate the interface.d.ts file which is only run at installation time.

To fix this missing functionality my plan is to update the regenerateAll() function to also generate the interface.d.ts file. Do you agree?

Alternatively, I can add the logic for generating the interface.d.ts file as part of the rosidl_gen/index.js#generateAll() function. 

Thoughts? 

I'll make the changes and test tomorrow.

Reply:
> To fix this missing functionality my plan is to update the regenerateAll() function to also generate the interface.d.ts file. Do you agree?

I agree we should update the generated interface when any change happens. What's more, as the `types/interfaces.d.ts` is generated dynamically, we'd better not to submit this into repo and it only exists locally. Thanks!

Reply:
fixed. See index.js#regenerateAll() which calls the tsd generator after generating the javascript code.